src/ast/conditions.js

```diff
@@ -0,0 +1,65 @@
+function isEqualTest ( node ) {
+	return node.type === 'BinaryExpression' && ( node.operator === '===' || node.operator === '==' );
+}
+
+function isNotEqualTest ( node ) {
+	return node.type === 'BinaryExpression' && ( node.operator === '!==' || node.operator === '!=' );
+}
+
+function nodesAreEqual ( a, b ) {
+	if ( a.type !== b.type ) return false;
+	if ( a.type === 'Literal' ) return a.value === b.value;
+	if ( a.type === 'Identifier' ) return a.name === b.name;
+
+	return false;
+}
+
+function nodesAreNotEqual ( a, b ) {
+	if ( a.type !== b.type ) return false;
+	if ( a.type === 'Literal' ) return a.value != b.value;
+	if ( a.type === 'Identifier' ) return a.name != b.name;
```

Comment:
yeah, it's to do with strict vs non-strict equality. though now that you mention it, i think it'd be better to have separate `nodesAreStrictEqual` and `nodesAreNotStrictEqual` functions